crates/sui-axelar-cgp/move/sources/messaging.move

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+module axelar::messaging {
+    use std::string;
+    use std::string::String;
+
+    #[test_only]
+    use std::vector;
+
+    /// Message struct which can consumed only by a `Channel` object.
+    /// Does not require additional generic field to operate as linking
+    /// by `id_bytes` is more than enough.
+    ///
+    /// Consider naming this `axelar::messaging::CallApproval`.
+    struct Message {
+        /// ID of the message, guaranteed to be unique by Axelar.
+        msg_id: vector<u8>,
+        /// The target Channel's UID.
+        target_id: address,
+        /// Name of the chain where this message came from.
+        source_chain: String,
+        /// Address of the source chain (vector used for compatibility).
+        /// UTF8 / ASCII encoded string (for 0x0... eth address gonna be 42 bytes with 0x)
+        source_address: String,
+        /// Hash of the full payload (including source_* fields).
+        payload_hash: vector<u8>,
+        /// The rest of the payload to be used by the application.
+        payload: vector<u8>,
```

Comment:
`payload` is not part of the message arriving at the gateway. We have only `payload_hash` there. 

The idea is not to put the whole message into the storage but rather only a hash of it's properties.

```
struct Message {
   approval_hash: vector<u8>,
}
```

Where
```
approval_hash = keccak256(msg_id, target_id, source_chain, source_address, payload_hash)
```
And then destination contract would need to accept all the props of the message but with the actual `payload` instead of `payload_hash`

Reply:
@damirka @patrickkuo

Reply:
one issue I have is unlike eth, our ec recover method need the full msg to recover the public key, @joyqvq can you advise if there are ways around this?